compiler/rustc_typeck/src/check/generator_interior.rs

```diff
@@ -290,7 +307,7 @@ impl<'a, 'tcx> Visitor<'tcx> for InteriorVisitor<'a, 'tcx> {
         if let PatKind::Binding(..) = pat.kind {
             let scope = self.region_scope_tree.var_scope(pat.hir_id.local_id);
             let ty = self.fcx.typeck_results.borrow().pat_ty(pat);
-            self.record(ty, Some(scope), None, pat.span, false);
+            self.record(ty, pat.hir_id, Some(scope), None, pat.span, false);
```

Comment:
Not your fault, but this list of arguments is getting out of hand... Maybe we should make it a builder pattern or just pass a struct with appropriate defaults, constructor functions or sth.